pkg/controllers/user/monitoring/clusterHandler.go

```diff
@@ -419,7 +420,17 @@ func (ch *clusterHandler) deployApp(appName, appTargetNamespace string, appProje
 		},
 	}
 
-	_, err = utils.DeployApp(ch.app.cattleAppClient, appDeployProjectID, app, false)
+	// redeploy cluster-monitoring App forcibly if its workload doesn't exist
+	var forceRedeploy bool
+	appWorkload, err := ch.app.agentStatefulSetClient.GetNamespaced(appTargetNamespace, fmt.Sprintf("prometheus-%s", appName), metav1.GetOptions{})
```

Comment:
why should we use agentStatefulSetClient and not lister here?

Reply:
I followed what monitoring-operator did with deploymentClient, but it should be okay to use lister, switched to lister